### ISSUE TITLE
Codec can not be used with the file input

### ISSUE THREAD

**Issue by purbon** [logstash-plugins/logstash-codec-gzip_lines#4]
This codec can not be used with the logstash file input because of the way it's implemented and how the file input works.

If we take this into consideration how this codec works at the current version:

```ruby
  def decode(data)
    @decoder = Zlib::GzipReader.new(data)

    begin
      @decoder.each_line do |line|
        yield LogStash::Event.new("message" => @converter.convert(line))
      end
    rescue Zlib::Error, Zlib::GzipFile::Error=> e
      file = data.is_a?(String) ? data : data.class

      @logger.error("Gzip codec: We cannot uncompress the gzip file", :filename => file)
      raise e
    end
  end # def decode
```

as we can see it expect it to have all data read at once and then it decompresses it to reach each line. But if we aim to use this codec with the file input it will not work because filewatch (internal file input library) is tailing each line individually before reaching the codec level.

This situation might change in future when [Mills](https://github.com/elastic/logstash/issues/4858) are introduced.

**Comment by ph** [comment]
I understand the issues now,  Before the `mills` concept land this problem need to be solved at the input level (i.e., file, s3, etc. ), the S3 input plugin already does that internally. Also in the current state of this plugin, I doubt it will even work with the output.

I don't think we ship this plugin by default, and I think we should just deprecate it, there is no way that this plugin can work in the current state of LS since we work with string and

**Comment by purbon** [comment]
@ph you are right, this is a community plugins. I opened the issue to track this situation and document the situation 😄 .

I think deprecation and removing might be a good idea, main motivation of me here was to reduce confusion and errors on user side.
